Add max bitrate option

utkarshdalal opened the pull request:
## Description
Added a parameter for max bitrate to the web UI, config file and `video.cpp`. Additionally, made some small changes to fix CI deployment script for Windows - renamed a conflicting parameter.


### Screenshot
<img width="1728" alt="image" src="https://github.com/LizardByte/Sunshine/assets/3746571/cfc9288b-d807-45e9-9440-835843a7d79d">


### Issues Fixed or Closed


## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [x] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LizardByte/Sunshine?pullRequest=1463) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LizardByte/Sunshine?pullRequest=1463) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LizardByte/Sunshine?pullRequest=1463) it.</sub>

ReenigneArcher commented on src/platform/windows/publish.cpp:
```diff
@@ -29,7 +29,7 @@ using namespace std::literals;
 
 extern "C" {
-constexpr auto DNS_REQUEST_PENDING = 9506L;
+constexpr auto MY_DNS_REQUEST_PENDING = 9506L;
```
We don't really name variables "my_variable".

utkarshdalal replied:
This was the change I had to make for the CI, it was failing with this error otherwise - https://github.com/utkarshdalal/Sunshine-GameAway/actions/runs/5632865670/job/15261219026. Screenshot below.
<img width="1728" alt="image" src="https://github.com/LizardByte/Sunshine/assets/3746571/95404174-08a8-4e0d-a6ae-55fd5bc7df55">

I can name it something else also if you like, but this variable name was conflicting with a reserved variable name.

ReenigneArcher commented on README.rst:
```diff
@@ -126,3 +126,4 @@ Stats
 .. _nvenc support matrix: https://developer.nvidia.com/video-encode-and-decode-gpu-support-matrix-new
 .. _obs-amd hardware support: https://github.com/obsproject/obs-amd-encoder/wiki/Hardware-Support
 .. _VAAPI hardware support: https://www.intel.com/content/www/us/en/developer/articles/technical/linuxmedia-vaapi.html
+
```
Undo this change, doesn't add any value

utkarshdalal replied:
Will do.

ReenigneArcher commented on src/config.cpp:
```diff
@@ -421,6 +421,7 @@ namespace config {
     },  // supported resolutions
 
     { 10, 30, 60, 90, 120 },  // supported fps
+    0,
```
Please add comment following the value

ReenigneArcher commented on src_assets/common/assets/web/config.html:
```diff
@@ -1051,6 +1064,7 @@ <h1 class="my-4">Configuration</h1>
     "vt_realtime": "enabled",
     "vt_software": "auto",
     "global_prep_cmd": "[]",
+    "max_bitrate": 0,
```
These should be alphabetical, please fix the one above yours while you're at it.

ReenigneArcher commented on src_assets/common/assets/web/config.html:
```diff
@@ -1159,6 +1173,7 @@ <h1 class="my-4">Configuration</h1>
           let resolutions = [];
           res.split(",").forEach((r) => resolutions.push(r.trim()));
           this.resolutions = resolutions;
+          this.max_bitrate = this.config.max_bitrate || 0;
```
We don't need this line for a simple int or str value. We just do it for more complex, json encoded values.

ReenigneArcher reviewed (changes requested):
Thanks for the PR. I have some changes to request. In addition to the other comments, we also need to update the advanced usage section of the docs.

ReenigneArcher commented:
Also, there was no change to the CI. Guessing you were working off the master branch, and fixed something we already fixed in nightly?

utkarshdalal commented:
@ReenigneArcher The PR is updated, the only thing that I didn't change was renaming the variable to `DNS_REQUEST_PENDING` from `MY_DNS_REQUEST_PENDING`. As I mentioned in the comment, doing this made my Windows build fail. Does this work for you? If so I can change it but I think renaming the variable is necessary.

ReenigneArcher commented on src/config.cpp:
```diff
@@ -421,6 +421,7 @@ namespace config {
     },  // supported resolutions
 
     { 10, 30, 60, 90, 120 },  // supported fps
+    0, // max bitrate
```
```suggestion
    0,  // max bitrate
```

2 spaces (linting rules)

ReenigneArcher commented on src_assets/common/assets/web/config.html:
```diff
@@ -200,6 +200,19 @@ <h1 class="my-4">Configuration</h1>
+            v-model="config.max_bitrate"
+          />
+          <div class="form-text">
+            Maximum bitrate for streaming in Kbps. If not specified, the default bitrate is used
```
```suggestion
            Maximum bitrate for streaming in Kbps.
```

We always use the default value if not specified (or if the value provided by the user is outside an acceptable range). Actually we should probably enforce a max value for the setting.

ReenigneArcher commented on src/config.cpp:
```diff
@@ -984,6 +985,7 @@ namespace config {
     list_string_f(vars, "resolutions"s, nvhttp.resolutions);
     list_int_f(vars, "fps"s, nvhttp.fps);
     list_prep_cmd_f(vars, "global_prep_cmd", config::sunshine.prep_cmds);
+    int_f(vars, "max_bitrate", nvhttp.max_bitrate);
```
```suggestion
    int_between_f(vars, "max_bitrate", nvhttp.max_bitrate, { 0, 150000 });
```

ReenigneArcher reviewed (changes requested):
I have a few more requested changes.

Regarding the variable, I think it should be changed back. I'm a bit confused why it failed for you as we have many builds going daily and not had any failures from this recently. My first inclination is that this was something that we fixed a while back in the nightly branch, but you were working from the master branch, so didn't have our fixes. (@cgutman thoughts?)

utkarshdalal commented:
@ReenigneArcher I've addressed all your comments including renaming the variable. Maybe you haven't seen it break because the action (Windows Build) only runs when pushing to the master branch. Anyway, whenever you want to make a new version, if it breaks changing the name of this variable is the fix.

ReenigneArcher commented:
> [![CLA assistant check](https://camo.githubusercontent.com/4c970efeed75dd01f6778bf7e33dec85e4171375cb69acb0927bfc9d437d78d9/68747470733a2f2f636c612d617373697374616e742e696f2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla-assistant.io/LizardByte/Sunshine?pullRequest=1463) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LizardByte/Sunshine?pullRequest=1463) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LizardByte/Sunshine?pullRequest=1463) it.

@utkarshdalal please sign the CLA if you want us to merge this. First comment in this PR has instructions. Thanks!

utkarshdalal commented:
@ReenigneArcher signed.

ReenigneArcher reviewed (changes requested):
I found an additional item to change. Placeholder should be the default value.

utkarshdalal commented on src_assets/common/assets/web/config.html:
```diff
@@ -200,6 +200,19 @@ <h1 class="my-4">Configuration</h1>
+            type="number"
+            class="form-control"
+            id="max_bitrate"
+            placeholder="5000"
```
It's done, sorry for the delay I missed the notification.

utkarshdalal commented:
Anything remaining to be done on this issue?

ReenigneArcher commented:
> Anything remaining to be done on this issue?

Agreement from other team members to merge this.

utkarshdalal commented:
I mean you guys aren't reviewing it so not sure what you want from me

On Sun, 31 Mar, 2024, 14:06 LizardByte-bot, ***@***.***>
wrote:

> It looks like this PR has been idle for 90 days. If it's still something
> you're working on or would like to pursue, please leave a comment or update
> your branch. Otherwise, we'll be closing this PR in 10 days to reduce our
> backlog. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LizardByte/Sunshine/pull/1463#issuecomment-2028624336>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4SWC4UXGUOAMDZHV4KMGDY27NZDAVCNFSM6AAAAAA2UO5LB2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRYGYZDIMZTGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ReenigneArcher commented:
@utkarshdalal I apologize, but at this time our team doesn't feel that this feature would be valuable to Sunshine. We can possibly revisit it in the future if there is more interest from users.
